### ISSUE TITLE
DeleteResult / UpdateResult not implemented for all Drivers

### ISSUE THREAD

**Issue by marko-asplund** [typeorm/typeorm#2415]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`

**TypeORM version:**

[ ] `latest`
[ ] `@next`
[x] `0.2.7` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**
1. invoke `Repository.delete`
2. the returned `DeleteResult` does not include any info at all. Expecting to include the number of affected rows.

I'm having trouble with `Repository.delete`: the operation successfully removes the entity from the database, but `DeleteResult` does not contain count of entities deleted, it's just `DeleteResult { raw: [] }`. This occurs with v0.2.7.

Also, there doesn't appear to be a tag for version 0.2.7 on GitHub though there's a release on npmjs. Why is that?

Why doesn't `DeleteResult` specify a property for the number of rows affected?

**Comment by pleerock** [comment]
> Also, there doesn't appear to be a tag for version 0.2.7 on GitHub though there's a release on npmjs. Why is that?

I'll fix that.

> Why doesn't DeleteResult specify a property for the number of rows affected?

because it wasn't implemented. We already have this feature reported, but noone still took on its implementation. Please feel free, its an easy feature to implement.

**Comment by marko-asplund** [comment]
@pleerock Can you give a reference to an earlier report? I just tried, but was unable to find one.

**Comment by pleerock** [comment]
I could find only #1308 which is related, need to search deeper to find that one. But anyway, you don't need that issue if you want to implement it. You know what your goal, you can easily do that. But please be sure to implement it for all drivers if you are going to do it. That's easy feature to implement

**Comment by thibaud-sanchez** [comment]
I have the same problem with SQLite and 0.2.8
It returns : `{"raw": []}`

**Comment by pleerock** [comment]
`raw` is just raw result returned by underlying database driver. As I pointed above we need to implement this feature request.

**Comment by HardySoftware** [comment]
MongoDb does not return anything, neither raw nor affected regardless if the record is found/deleted or not.

**Comment by akosasante** [comment]
In case it helps anyone else in the future, for postgres I ended up just using 
```typescript
       await this.db.createQueryBuilder()
            .delete()
            .whereInIds(id)
            .returning("id")
            .execute();
```

It returns a DeleteResult like: `{ raw: [ {id: 1} ], affected: 1 }`. You could also do `.returning("*")` if you want the full object returned.

**Comment by varogonz95** [comment]
Is this implemented? I'm facing this issue too with sqlite3 and `typeorm@^0.2.12`

**Comment by dennbagas** [comment]
> ```ts
>        await this.db.createQueryBuilder()
>             .delete()
>             .whereInIds(id)
>             .returning("id")
>             .execute();
> ```

this didn't work for me, it just returns an empty object. I use typeorm@0.2.18.

**Comment by liv1n9** [comment]
> MongoDb does not return anything, neither raw nor affected regardless if the record is found/deleted or not.

Did it fixed?

**Comment by liv1n9** [comment]
> MongoDb does not return anything, neither raw nor affected regardless if the record is found/deleted or not.

Has it been fixed?

**Comment by VinayKrMittal** [comment]
> **Issue type:**
> 
> [ ] question
> [x] bug report
> [ ] feature request
> [ ] documentation issue
> 
> **Database system/driver:**
> 
> [ ] `cordova`
> [ ] `mongodb`
> [ ] `mssql`
> [ ] `mysql` / `mariadb`
> [ ] `oracle`
> [x] `postgres`
> [ ] `sqlite`
> [ ] `sqljs`
> [ ] `react-native`
> 
> **TypeORM version:**
> 
> [ ] `latest`
> [ ] `@next`
> [x] `0.2.7` (or put your version here)
> 
> **Steps to reproduce or a small repository showing the problem:**
> 
> 1. invoke `Repository.delete`
> 2. the returned `DeleteResult` does not include any info at all. Expecting to include the number of affected rows.
> 
> I'm having trouble with `Repository.delete`: the operation successfully removes the entity from the database, but `DeleteResult` does not contain count of entities deleted, it's just `DeleteResult { raw: [] }`. This occurs with v0.2.7.
> 
> Also, there doesn't appear to be a tag for version 0.2.7 on GitHub though there's a release on npmjs. Why is that?
> 
> Why doesn't `DeleteResult` specify a property for the number of rows affected?

same issue for me. any solution..

**Comment by in7rude** [comment]
Same issue. Deleting record in MongoDB results an empty DeleteResult object

**Comment by danielb7390** [comment]
Trying to change from sequelize... and i hit this "bug", quite sad this issue is still open..

**Comment by TBG-FR** [comment]
@AndreySavenkov  Same here. I'm curious, how did you managed that then ?

**Comment by adelsagemilang** [comment]
Is there any alternative solution for this?

> Same issue. Deleting record in MongoDB results an empty DeleteResult object

**Comment by ynwd** [comment]
> Same issue

sqlite

**Comment by isdiop** [comment]
> > MongoDb does not return anything, neither raw nor affected regardless if the record is found/deleted or not.
> 
> Has it been fixed?

+1

**Comment by TBG-FR** [comment]
@imnotjames can you also add `driver:mongodb` (or should we open another issue ?)

**Comment by viniciusmolina12** [comment]
Same issue in mssql driver

**Comment by Juraj-Masiar** [comment]
This should be fixed also for the "INSERT" where Postgres returns number of inserted rows.
I'm using `.onConflict('do nothing')` so I now don't know if the row was inserted or not. At least here I can use `.returning('id')` as a workaround.

**Comment by gangss2** [comment]
Not fixed to SQLite3?

`UpdateResult { generatedMaps: [], raw: [] }`

**Comment by nebkat** [comment]
Am proposing that processing of results is moved to individual drivers as part of #7058  https://github.com/typeorm/typeorm/pull/7058/commits/e666d7d515bdaaa6c85f693929b6ed1371ebb812.

**Comment by SalahAdDin** [comment]
Two years and same problem with **Mongo**: it always returns `{}`, it does not matter if it deletes the user or not, it does not return any error.

**Comment by Zyles** [comment]
Why is this not a priority?

**Comment by Kolobamanacas** [comment]
So, am I correct: there is no way to check whether any rows have been affected by delete method for PostgreSQL, right?

**Comment by matomesc** [comment]
@Kolobamanacas this works as expected for postgres on `0.2.32`:

```
// Deleting one row
const result = await repo.delete(userA.id);
// Return result is
DeleteResult { raw: [], affected: 1 }
```

**Comment by nbaua** [comment]
No one really looking in to UpdateResult or DeleteResult for mongodb, its been a while this thread is still open..

**Comment by VictorLM** [comment]
Still getting empty result using MongoDB driver.

**Comment by imnotjames** [comment]
The logic for query results have been moved into the drivers via https://github.com/typeorm/typeorm/pull/7753

This means implementing it is up to the drivers.

The MongoDB driver doesn't return lots of this because it's EXTREMELY janky.  It was implemented to query by not really doing things like any other driver & hacks up the whole TypeORM library to make it barely work..

Might still be able to make it sort of work, though, if the underlying module supports it.

Even then, though, it might not help much because it doesn't use the query builder and all that jazz.

**Comment by imnotjames** [comment]
This should be implemented for every driver now except for MongoDB with #7753 - to be in the next release.

**Comment by imnotjames** [comment]
Opened a PR to implement it for the mongo EntityManager because that seems to be where that logic lives..

**Comment by marcosdipaolo** [comment]
And for the rest of the drivers? sqlite here, DeleteResult doesn't return the proper affected value

**Comment by imnotjames** [comment]
> And for the rest of the drivers? sqlite here, neither DeleteResult nor UpdateResult have the right info

With the master branch?

**Comment by marcosdipaolo** [comment]
> > And for the rest of the drivers? sqlite here, neither DeleteResult nor UpdateResult have the right info
> 
> With the master branch?

"typeorm": "^0.2.34",

**Comment by imnotjames** [comment]
This code has not been released yet.  We're aiming for a release this week.

**Comment by jdeg** [comment]
I'm still having this issue in SQLite3. Any news about it?

**Comment by imnotjames** [comment]
> I'm still having this issue in SQLite3. Any news about it?

Should be released at this point.  If it's not working for your use case please open a new issue.

**Comment by HazemHa** [comment]
it is not solved yet
MySQL version: 8.0.27

but you can do it by yourself


```
async remove(id: string): Promise<DeleteResult> {
        const deletedRow = await this.nodesRepository.findOne({ uuid: id });
        const deleteResultNode = await this.nodesRepository.delete({ uuid: id })
        if (deleteResultNode.affected > 0) {
            deleteResultNode.raw.push(deletedRow)
        }

        return deleteResultNode;

    }
```

**Comment by DanielYoung-Till** [comment]
2024 now.

**Comment by SteTSC** [comment]
> 2024 now.

As noted by @DanielYoung-Till this is still not resolved in 0.3.20.

Code:
```
  update(myEntity: MyEntity): Promise<MyEntity> {
    return this.repository
      .update({ id: myEntity.id }, myEntity)
      .then((event) => {
        console.log(event);
      });
```

Expected:
```
{ generatedMaps: [], raw: [], affected: 1 }
```
Actual:
```
{ generatedMaps: [], raw: [{ id:123, ...etc }], affected: 1 }
```
